dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/GitRepositoryDetector.kt

```diff
@@ -39,6 +41,32 @@ internal class GitRepositoryDetector : RepositoryDetector {
     }
 
     // region Internal
+    @Suppress("SwallowedException")
+    private fun sanitizeUrl(remoteUrl: String): String {
+        return try {
+            val parsedUri = URI(remoteUrl)
+            if (parsedUri.userInfo.isNullOrEmpty()) {
+                parsedUri.toString()
+            } else {
+                val sanitizedUri = URI(
+                    parsedUri.scheme,
+                    null,
+                    parsedUri.host,
+                    parsedUri.port,
+                    parsedUri.path,
+                    parsedUri.query,
+                    parsedUri.fragment
+                )
+                sanitizedUri.toString()
+            }
+        } catch (e: URISyntaxException) {
+            LOGGER.warn(
+                "The detected GIT remote url: [$remoteUrl] is not of type SSL." +
```

Comment:
the interesting point is that per https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols SSH remote URL can be one of the 2 formats:

* `ssh://[user@]server/project.git`
* or SCP-like syntax `[user@]server:project.git` (I missed it when first checked the page)

So former case should be handled fine by the branch above (with `URI` parsing), but latter case won't be and it still should be sanitized - I'm not expert in SCP remote format, but even if it allows just to specify only user name without a password, it is still a sensitive info.

Reply:
Oh...that's interesting, was actually wondering if the user information could be provided through SSH either. In this case I am not sure how I could avoid using `RegEx` to strip that part...not sure if I could find an SSH parser in Java, will have a look. Problem with `RegEx` is that it might not work if the format is not exactly the one you expect it to be and the question is : if we use `RegEx` for SSH then maybe we should have a rule that can be applied for both SSL or SSH formats and not use the `URI` parser anymore. What do you both think ? @xgouchet @0xnm

Reply:
I would say that we need only browsable remote url, which is only `http[s]://` schema and not `ssh://`, `git://` or whatever it is, even schema-less like in case of SCP syntax. Currently we send whatever is the output of the `git remote get-url origin` command. But maybe we can simplify it? Like don't send anything what doesn't start with `http[s]://` and if it should be sent, then sanitize it (which limits it to the URLs starting with `http[s]://` then).

As I understand, the point of this feature is to just have a browsable link, don't think we are interested in non-HTTP URLs.

Reply:
Yes...reason why I keep it like that is that in the JS parser they actually have a `try/catch` mechanism and in case the url is not of `http[s]://(.+)` format they are just returning the unparsed URL. But as @xgouchet stated also there I think we should not allow at all non `http[s]` formats and we can even fail the build here with an error message.

Reply:
I will update the PR accordingly

Reply:
Not sure we should fail a build. I'm using `git@github.com:DataDog/dd-sdk-android.git` for `dd-sdk-android` and it is perfectly valid URL, I would be a bit disappointed if something in the build doesn't like it. Maybe it may give just a warning instead.

If we fail the build, it may be CI build, which will add a lot of burden for the people watching it.

Reply:
Also it may be a company policy to use SSH-like URLs with cert auth instead of HTTPS, so they cannot change it.

Reply:
Yes...I think I will just send a warning message, also please note that there is another PR (draft for now) which I just opened where we are letting the client to customize the `gitRemoteUrl` through the plugin extension. This will solve the case when you are using `SSH` . In conclusion using a warn message here will make the user aware that he needs to set an HTTP[S] url format manually to have the plugin working correctly.

Reply:
We need to be able to support standard remote url formats, and SSH urls `git@…` **are** a standard (it's used at least by Github, Gitlab, Bitbucket, …). The end goal is to be able to create a browsable link from a crash report but the backend, AFAIK, can convert the URIs in standard formats into a url (e.g.: `git@github.com:user/repository.git` → `https://github.com/user/repository/`).

Reply:
Oh...this changes the approach then, so in this case we will keep that URL and probably sanitize it differently (through a RegEx maybe) but still send it to the backend after. Do we agree on this @xgouchet @0xnm

Reply:
One option to do this without having to write a url compliant regex (which is going to be complex and tedious to write and test) is to search for the `"://"` substring to check wether or not the URI has a scheme. If not, you can prepend the string with `"ssh://"` and it will be parsed accordingly.

Reply:
Hmmm the last option seems more reliable and I can apply the same logic after by using the URI. I will try to go with this approach.

Reply:
I wasn't aware about this backend feature, that it can do the conversion. The interesting part is that if remote is of SCP syntax, like `ssh://user@server:project.git`, then adding `ssh://` helps to parse it, but `getUserInfo` returns `null` https://ideone.com/eUkYAV